core/arch/aarch64/emit_utils.c

```diff
@@ -213,15 +214,15 @@ insert_exit_stub_other_flags(dcontext_t *dcontext, fragment_t *f, linkstub_t *l,
         /* The final slot is a data slot, which will hold the address of either
          * the fcache-return routine or the linked fragment.
          */
+        ASSERT(pc == target_pc_slot || pc + 1 == target_pc_slot);
         ASSERT(sizeof(app_pc) == 8);
         pc += DIRECT_EXIT_STUB_DATA_SZ / sizeof(uint);
         /* We start off with the fcache-return routine address in the slot. */
-        /* AArch64 uses shared fragments, so fcache_return gencode should be
-         * shared too.
+        /* AArch64 uses shared gencode. So, fcache_return routine address should be
+         * same, no matter which thread creates/unpatches the stub.
          */
         ASSERT(fcache_return_routine(dcontext) == fcache_return_routine(GLOBAL_DCONTEXT));
-        *(uint64 *)get_target_pc_slot(f, stub_pc) =
-            (ptr_uint_t)fcache_return_routine(dcontext);
+        *(uint64 *)target_pc_slot = (ptr_uint_t)fcache_return_routine(dcontext);
```

Comment:
See comment about treating target_pc_slot as uint*: this uint64* makes more sense.  But it seems odd here to have uint64* on one side, but ptr_uint_t on the other.

Reply:
LHS is `ptr_uint_t*` too now.